main/src/visualization/plotPerf.js

```diff
@@ -24,114 +34,38 @@ const options = {
   };
-  const width = 960 - labelMargin.left - margin.left - margin.right;
-  const height = 500 - labelMargin.bottom - margin.top - margin.bottom;
-
-  const maxFps = d3.max(instantFPS);
-  const xOtherScale = d3.scaleLinear()
-    .domain([0, frameTimes[frameTimes.length - 1] - frameTimes[0]])
-    .range([0, width]);
-
-  const x = d3.scaleLinear()
-    .domain([frameTimes[0] - timeBetweenFrames[0], frameTimes[frameTimes.length - 1]])
-    .range([0, width]);
-
-  const y = d3.scaleLinear()
-    .domain([0, d3.max(instantFPS)])
-    .range([height, 0]);
-
-
-  const svgWidth = width + labelMargin.left + margin.left + margin.right;
-  const svgHeight = height + labelMargin.bottom + margin.top + margin.bottom;
-
-  function getXOffset(data, index) {
-    return x(data - timeBetweenFrames[index]);
-  }
-
-  function getYOffset(index) {
-    return y(instantFPS[index]);
-  }
-
-  const svg = d3n.createSVG(svgWidth, svgHeight)
-    .append('g')
-    .attr('transform', `translate(${margin.left},${margin.top})`);
-
-  // Label Names
-  // text label for the x axis
-  svg.append('text')
-    .attr(
-      'transform',
-      `translate(${width / 2} ,${
-      height + margin.top + 30})`,
-  )
-    .style('text-anchor', 'middle')
-    .text('time since Origin in ms');
-
-  // text label for the y axis
-  svg.append('text')
-    .attr('transform', 'rotate(-90)')
-    .attr('y', 0)
-    .attr('x', 0 - (height / 2))
-    .attr('dy', '1em')
-    .style('text-anchor', 'middle')
-    .text('FPS');
-
-
-  // With label Margins
-  const graph = svg
-    .append('g')
-    .attr('transform', `translate(${labelMargin.left}, 0)`);
-  const bar = graph.selectAll('.bar')
-    .data(frameTimes)
-    .enter().append('g')
-    .attr('class', 'bar')
-    .attr('transform', (d, i) => `translate(${getXOffset(d, i)},${getYOffset(i)})`);
-
-  bar.append('rect')
-    .attr('width', (d, i) => xOtherScale(timeBetweenFrames[i]))
-    .attr('height', (d, i) => height - getYOffset(i));
-
-  const dragEventsSVG = graph.selectAll('.drag')
-    .data(dragEvents)
-    .enter().append('g')
-    .attr('class', 'drag')
-    .attr('transform', d => `translate(${x(d.start)},${height - DRAG_EVENTS_HEIGHT})`);
-
-  dragEventsSVG.append('rect')
-    .attr('width', d => x(d.end) - x(d.start))
-    .attr('height', () => DRAG_EVENTS_HEIGHT);
-  dragEventsSVG.append('text')
-    .attr('dy', '.75em')
-    .attr('y', (DRAG_EVENTS_HEIGHT / 2) - 5)
-    .attr('x', d => xOtherScale((d.end - d.start) / 2))
-    .attr('text-anchor', 'middle')
-    .text((d, i) => `DragEvent${i}`);
-
-  graph.append('g')
-    .attr('class', 'axis axis--x')
-    .attr('transform', `translate(0,${height})`)
-    .call(d3.axisBottom(x));
-
-  graph.append('g')
-    .attr('class', 'axis axis--y')
-    .call(d3.axisLeft(y));
-  return d3n.svgString();
+  const svgWidth = 960;
+  const svgHeight = 500;
+
+  const svgBuilderObj = new svgBuilder.SVGBuilder(svgWidth, svgHeight, margin, labelMargin, options);
```

Comment:
Minor: When you have a module that only contains one class, you may want to export directly the class, this way the code here could be a bit simpler:
`const svgBuilder = new SVGBuilder(...);`
Alternatively, if you have the same case but multiple exports, you can also import the class only, e.g.
`const SVGBuilder = require('./svgBuilder').SVGBuilder;`
(and then use the same code above for instantiating it)

(the same also applies for the export of `SVGFromLogs`)